components/script/dom/document.rs

```diff
@@ -1975,6 +1978,16 @@ impl Document {
                            ReflowQueryType::NoQuery,
                            ReflowReason::ElementStateChanged);
     }
+
+    pub fn serialize(&self) -> Fallible<DOMString> {
+        let mut writer = vec![];
+        if let Ok(()) = serialize(&mut writer, &self.upcast::<Node>(),
+            SerializeOpts { traversal_scope: ChildrenOnly, ..Default::default() }) {
```

Comment:
Just answering this. If I don't use it, the serializer will break with this message.
```
  ▶ CRASH [expected OK] /XMLHttpRequest/send-entity-body-document.htm
  │ 
  │ Can't serialize Document node itself (thread ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(0) }, at /home/jayson/p/servo/components/script/parse/html.rs:244)
  │ stack backtrace:
  │    0:     0x559b20a9ed9d - backtrace::backtrace::trace::h0e60ef08c7c34e9f
  │    1:     0x559b20a9f422 - backtrace::capture::Backtrace::new::h8bf319c36d8f5d1b
  │    2:     0x559b1f1dc19d - servo::main::_{{closure}}::hd8627710ce441e59
  │    3:     0x559b20cf21e5 - std::panicking::rust_panic_with_hook::hb1322e5f2588b4db
  │    4:     0x559b1f5d54bf - std::panicking::begin_panic::h0914615a412ba184
  │    5:     0x559b1fb71014 - script::parse::html::_<impl html5ever..serialize..Serializable for &'a script..dom..node..Node>::serialize::hb19cf3d1a8128106
  │    6:     0x559b1f827c4e - html5ever::serialize::serialize::h5ca983e95371f52f
  │    7:     0x559b1f9e2530 - script::dom::document::Document::serialize::h29b3a36f0c434859
  │    8:     0x559b1fb63be0 - _<script..dom..bindings..codegen..UnionTypes..DocumentOrBodyInit as script..dom..xmlhttprequest..Extractable>::extract::h6808393b4e7b8052
  │    9:     0x559b1fb5bcd3 - _<script..dom..xmlhttprequest..XMLHttpRequest as script..dom..bindings..codegen..Bindings..XMLHttpRequestBinding..XMLHttpRequestBinding..XMLHttpRequestMethods>::Send::h337db624ae2e3ae8
  │   10:     0x559b1f5db6f3 - std::panicking::try::do_call::h0608cafe54224437
  │   11:     0x559b20cfa456 - __rust_maybe_catch_panic
  │   12:     0x559b1f999600 - script::dom::bindings::codegen::Bindings::XMLHttpRequestBinding::XMLHttpRequestBinding::send::hf8d32e9e3f3c3781
  │   13:     0x559b1fd1d184 - CallJitMethodOp
  │   14:     0x559b1f862854 - script::dom::bindings::utils::generic_call::hc239be2aefa6c7d4
  │   15:     0x559b200e2942 - 2js12CallJSNativeEP9JSContextPFbS1_jPN2JS5ValueEERKNS2_8CallArgs
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/jscntxtinlines.h:232
  │                          - 2js23InternalCallOrConstructEP9JSContextRKN2JS8CallArgsENS_14MaybeConstruct
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/vm/Interpreter.cpp:453
  │   16:     0x559b200de0f1 - 2js13CallFromStackEP9JSContextRKN2JS8CallArgs
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/vm/Interpreter.cpp:504
  │                          - Interpret
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/vm/Interpreter.cpp:2873
  │   17:     0x559b200e265f - 2js9RunScriptEP9JSContextRNS_8RunState
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/vm/Interpreter.cpp:399
  │   18:     0x559b200e2880 - 2js23InternalCallOrConstructEP9JSContextRKN2JS8CallArgsENS_14MaybeConstruct
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/vm/Interpreter.cpp:471
  │   19:     0x559b200e2b47 - 2js4CallEP9JSContextN2JS6HandleINS2_5ValueEEES5_RKNS_13AnyInvokeArgsENS2_13MutableHandleIS4_E
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/vm/Interpreter.cpp:517
  │   20:     0x559b1ffd0edc - 2js9fun_applyEP9JSContextjPN2JS5Value
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/jsfun.cpp:1325
  │   21:     0x559b200e2942 - 2js12CallJSNativeEP9JSContextPFbS1_jPN2JS5ValueEERKNS2_8CallArgs
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/jscntxtinlines.h:232
  │                          - 2js23InternalCallOrConstructEP9JSContextRKN2JS8CallArgsENS_14MaybeConstruct
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/vm/Interpreter.cpp:453
  │   22:     0x559b200de0f1 - 2js13CallFromStackEP9JSContextRKN2JS8CallArgs
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/vm/Interpreter.cpp:504
  │                          - Interpret
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/vm/Interpreter.cpp:2873
  │   23:     0x559b200e265f - 2js9RunScriptEP9JSContextRNS_8RunState
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/vm/Interpreter.cpp:399
  │   24:     0x559b200e2880 - 2js23InternalCallOrConstructEP9JSContextRKN2JS8CallArgsENS_14MaybeConstruct
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/vm/Interpreter.cpp:471
  │   25:     0x559b200e2b47 - 2js4CallEP9JSContextN2JS6HandleINS2_5ValueEEES5_RKNS_13AnyInvokeArgsENS2_13MutableHandleIS4_E
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/vm/Interpreter.cpp:517
  │   26:     0x559b1ff6650a - _Z20JS_CallFunctionValueP9JSContextN2JS6HandleIP8JSObjectEENS2_INS1_5ValueEEERKNS1_16HandleValueArrayENS1_13MutableHandleIS6_EE
  │                         at /home/jayson/p/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/master/mozjs/js/src/jsapi.cpp:2781
  │   27:     0x559b1f8b8c1e - script::dom::bindings::codegen::Bindings::EventHandlerBinding::EventHandlerNonNull::Call::hfff992aab4ab8b4e
  │   28:     0x559b1f8b88c5 - script::dom::bindings::codegen::Bindings::EventHandlerBinding::EventHandlerNonNull::Call_::h86d2e00a2172acc3
  │   29:     0x559b1fa1790a - script::dom::eventdispatcher::handle_event::h16053516d9ef050b
  │   30:     0x559b1fa19c95 - script::dom::eventdispatcher::invoke::h1eae724394b1a36d
  │   31:     0x559b1fa18fef - script::dom::eventdispatcher::dispatch_event::h44cedf88bf8e65be
  │   32:     0x559b1fa1c9f2 - script::dom::eventtarget::EventTarget::fire_event::hc088fba0976b8daa
  │   33:     0x559b1fa622dc - script::dom::htmliframeelement::HTMLIFrameElement::iframe_load_event_steps::hb6a68ea29c7054a6
  │   34:     0x559b1fb8d5d2 - script::script_thread::ScriptThread::handle_msg_from_constellation::h4e39a1d4f2af6997
  │   35:     0x559b1fbd8b01 - script::script_thread::ScriptThread::handle_msgs::_{{closure}}::had1422a31c223a9c
  │   36:     0x559b1fb86de4 - script::script_thread::ScriptThread::handle_msgs::he0c72778ea94b501
  │   37:     0x559b1f631517 - std::panicking::try::do_call::h4c43a18df829bea5
  │   38:     0x559b20cfa456 - __rust_maybe_catch_panic
  │   39:     0x559b1f7943e4 - _<F as alloc..boxed..FnBox<A>>::call_box::h9939d2c8ad214242
  │   40:     0x559b20cf0820 - std::sys::thread::Thread::new::thread_start::h022e3887023c6290
  │   41:     0x7f9b8a9b36f9 - start_thread
  │   42:     0x7f9b8a4d3b5c - clone
  │   43:                0x0 - <unknown>
  │ ERROR:servo: Can't serialize Document node itself
  └ Pipeline failed in hard-fail mode.  Crashing!
```